Nos/Models/Persistence.swift

```diff
@@ -228,14 +228,20 @@ class PersistenceController {
                     let receivedAt = firstEventToDelete.receivedAt {
                     deleteBefore = receivedAt
                 }
-                   
+
+                let oldStoryCutoff = Calendar.current.date(byAdding: .day, value: -2, to: .now) ?? .now
+
                 // Delete events older than `deleteBefore`
                 let oldEventsRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "Event")
                 oldEventsRequest.sortDescriptors = [NSSortDescriptor(keyPath: \Event.receivedAt, ascending: true)]
+                let oldEventClause = "(receivedAt <= %@ OR receivedAt == nil)"
+                let notOwnEventClause = "(author.hexadecimalPublicKey != %@)"
+                let readStoryClause = "(isRead = 1 AND receivedAt > %@)"
```

Comment:
I think this will keep all events for the last 2 days, which I'm slightly concerned about because sometimes people leave the Discover tab open on the Mac and it can run for hours and download hundreds of thousands of events. That's a separate problem we need to solve, but right now you can always make the app fast again by restarting it, and I think that's a good property to preserve. So adding something like `ANY author.followers.source = %@` to this clause would let us delete events from people we don't follow.

Reply:
Looks like I misread the code, I think this is doing the right thing now.